api/transport/router.go

```diff
@@ -25,33 +25,25 @@ import "context"
 // TODO: Until golang/mock#4 is fixed, imports in the generated code have to
 // be fixed by hand. They use vendor/* import paths rather than direct.
 
-//go:generate mockgen -destination=transporttest/register.go -package=transporttest go.uber.org/yarpc/api/transport Registry,Registrar
+//go:generate mockgen -destination=transporttest/router.go -package=transporttest go.uber.org/yarpc/api/transport Router,RouteTable
 
-// ServiceProcedure represents a service and procedure registered against a
-// Registry.
-type ServiceProcedure struct {
-	Service   string
-	Procedure string
-}
-
-// Registrant specifies a single handler registered against the registry.
-type Registrant struct {
-	// Service name or empty to use the default service name.
-	Service string
+// Procedure specifies a single handler registered in the RouteTable.
+type Procedure struct {
+	// ServiceName or empty to use the default service name.
+	ServiceName string
 
-	// Name of the procedure.
-	Procedure string
+	// ProcedureName of the procedure.
+	ProcedureName string
 
 	// HandlerSpec specifiying which handler and rpc type.
 	HandlerSpec HandlerSpec
 }
 
-// Registry maintains and provides access to a collection of procedures and
-// their handlers.
-type Registry interface {
-	// ServiceProcedures returns a list of services and their procedures that
+// Router maintains and provides access to a collection of procedures
+type Router interface {
+	// Procedures returns a list of services and their procedures that
```

Comment:
No longer returning ServiceProcedures. Just say you're returning all
Procedures that this Router knows about. The concept of registering doesn't
exist on the Router, but the RouteTable.